wgpu-core/src/pipeline.rs

```diff
@@ -18,6 +18,7 @@ pub enum ShaderModuleSource<'a> {
 #[cfg_attr(feature = "replay", derive(serde::Deserialize))]
 pub struct ShaderModuleDescriptor<'a> {
     pub label: Label<'a>,
+    pub runtime_checks: bool,
```

Comment:
wgpu-core also has `unsafe`-sensitive API. This needs to be unsafe at this level, too.
Maybe wgpu-types can have `struct ShaderBoundChecks` as an opaque type, and we'll add `unsafe fn unchecked() -> Self` constructor?

Reply:
And how would the backends query if it's unchecked, trough a method?

Reply:
when wgpu-core gets this info and passes to wgpu-hal, it can do it as a `bool`, since wgpu-hal is unsafe.

Reply:
I've updated the PR with your suggestion could you check if it's how you'd expect